src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -0,0 +1,899 @@
+        scope = vnet_subnet_id
+        if not _add_role_assignment(cmd.cli_ctx, 'Network Contributor', service_principal, scope=scope):
+            logger.warning('Could not create a role assignment for subnet. '
+                           'Are you an Owner on this subscription?')
+
+    network_profile = None
+    if any([network_plugin, pod_cidr, service_cidr, dns_service_ip, docker_bridge_address]):
+        network_profile = ContainerServiceNetworkProfile(
+            network_plugin=network_plugin,
+            pod_cidr=pod_cidr,
+            service_cidr=service_cidr,
+            dns_service_ip=dns_service_ip,
+            docker_bridge_cidr=docker_bridge_address
+        )
+
+    addon_profiles = _handle_addons_args(
+        cmd,
+        enable_addons,
+        subscription_id,
+        resource_group_name,
+        {},
+        workspace_resource_id
+    )
+    if 'omsagent' in addon_profiles:
+        _ensure_container_insights_for_monitoring(cmd, addon_profiles['omsagent'])
+    aad_profile = None
+    if any([aad_client_app_id, aad_server_app_id, aad_server_app_secret, aad_tenant_id]):
+        aad_profile = ManagedClusterAADProfile(
+            client_app_id=aad_client_app_id,
+            server_app_id=aad_server_app_id,
+            server_app_secret=aad_server_app_secret,
+            tenant_id=aad_tenant_id
+        )
+
+    # Check that both --disable-rbac and --enable-rbac weren't provided
+    if all([disable_rbac, enable_rbac]):
+        raise CLIError('specify either "--disable-rbac" or "--enable-rbac", not both.')
+
+    mc = ManagedCluster(
+        location=location, tags=tags,
+        dns_prefix=dns_name_prefix,
+        kubernetes_version=kubernetes_version,
+        enable_rbac=False if disable_rbac else True,
+        agent_pool_profiles=[agent_pool_profile],
+        linux_profile=linux_profile,
+        service_principal_profile=service_principal_profile,
+        network_profile=network_profile,
+        addon_profiles=addon_profiles,
+        aad_profile=aad_profile)
+
+    # Due to SPN replication latency, we do a few retries here
+    max_retry = 30
+    retry_exception = Exception(None)
+    for _ in range(0, max_retry):
+        try:
+            return sdk_no_wait(no_wait, client.managed_clusters.create_or_update,
+                               resource_group_name=resource_group_name, resource_name=name, parameters=mc)
+        except CloudError as ex:
+            retry_exception = ex
+            if 'not found in Active Directory tenant' in ex.message:
+                time.sleep(3)
+            else:
+                raise ex
+    raise retry_exception
+
+
+def aks_update(cmd, client, resource_group_name, name, enable_cluster_autoscaler=False,
+               disable_cluster_autoscaler=False,
+               update_cluster_autoscaler=False,
+               min_count=None, max_count=None, no_wait=False):
+    if (all([enable_cluster_autoscaler, disable_cluster_autoscaler]) or
```

Comment:
I believe lines 486 to 495 can be simplified to:

```
if ( all([enable_cluster_autoscaler, disable_cluster_autoscaler, update_cluster_autoscaler]) or not(enable_cluster_autoscaler ^ disable_cluster_autoscaler ^ update_cluster_autoscaler))
            raise CLIError('specify either "--enable_cluster_autoscaler" or '
                       '"--disable_cluster_autoscaler" or '
                       '"--update_cluster_autoscaler".')
```

[see boolean table of 3 var xor](https://i.stack.imgur.com/apqpw.png)

this might be messier or you can split it into 2 ifs. its a matter of style so its more of a suggestion.

Reply:
youre approach is also valid, just wanted to play with xors :+1:

Reply:
Putting in list/tuple and counting is probably the simplest method